ConversorMedidas/app/src/main/java/projetointegrado/com/conversormedidas/ConversorDistancia.java

```diff
@@ -20,11 +21,26 @@ private String formatarNumero(Double numero){
         BigDecimal value = new BigDecimal(numero);
         BigDecimal grande = new BigDecimal(1000000000); // Teto de valor para não notação científica
         BigDecimal pequeno = new BigDecimal(0.001); // Chão
+        BigDecimal zero = new BigDecimal(0);
+
+
+        if(numero == 0){
+            return String.valueOf(0);
+        }
         if(value.compareTo(grande) > 0 || value.compareTo(pequeno) < 0){
             df = new DecimalFormat(
                     "0.00E0",
                     new DecimalFormatSymbols(new Locale("pt", "BR")));
         }
+        if(df.format((value.floatValue())).contentEquals("0")){
```

Comment:
If the rounded value is 0, it probably has some issues, probably a very small number near zero, and that's where the bug was treated